pandas/tests/frame/methods/test_diff.py

```diff
@@ -169,3 +169,41 @@ def test_diff_sparse(self):
         )
 
         tm.assert_frame_equal(result, expected)
+
+    def test_diff_integer_na(self):
+        # GH#24171 IntegerNA Support for DataFrame.diff()
+        df = pd.DataFrame(
+            {
+                "a": np.repeat([0, 1, np.nan, 2], 2),
+                "b": np.tile([0, 1, np.nan, 2], 2),
+                "c": np.repeat(np.nan, 8),
+                "d": np.arange(1, 9) ** 2,
+            },
+            dtype="Int64",
+        )
+
+        # Test case for default behaviour of diff
+        result_default = df.diff()
```

Comment:
hate to be pendantic about this, but can you use result & expected for both cases (the variable names)

Reply:
Sure. Didn't want to overwrite the variable names here. Would you suggest to overwrite, or create a separate test function?

Reply:
just overwriting the same names is fine here

Reply:
I parameterized the test so overwriting is not necessary anymore.

Reply:
> I parameterized the test so overwriting is not necessary anymore.

Did you push these changes? Currently I can't see any parametrisation